src/Umbraco.Web.BackOffice/Controllers/UsersController.cs

```diff
@@ -661,21 +666,38 @@ public async Task<ActionResult<ModelWithNotifications<string>>> PostChangePasswo
                 return new ValidationErrorResult(new SimpleValidationModel(ModelState.ToErrorDictionary()));
             }
 
-            var intId = changingPasswordModel.Id.TryConvertTo<int>();
+            Attempt<int> intId = changingPasswordModel.Id.TryConvertTo<int>();
             if (intId.Success == false)
             {
                 return NotFound();
             }
 
-            var found = _userService.GetUserById(intId.Result);
+            IUser found = _userService.GetUserById(intId.Result);
             if (found == null)
             {
                 return NotFound();
             }
 
-            // TODO: Why don't we inject this? Then we can just inject a logger
-            var passwordChanger = new PasswordChanger(_loggerFactory.CreateLogger<PasswordChanger>());
-            var passwordChangeResult = await passwordChanger.ChangePasswordWithIdentityAsync(_backofficeSecurityAccessor.BackOfficeSecurity.CurrentUser, found, changingPasswordModel, _userManager);
+            IUser currentUser = _backofficeSecurityAccessor.BackOfficeSecurity.CurrentUser;
+
+            // if it's the current user, the current user cannot reset their own password
+            if (currentUser.Username == found.Username)
+            {
+                return new ValidationErrorResult("Password reset is not allowed");
+            }
+
+            // if the current user has access to reset/manually change the password
+            if (currentUser.HasSectionAccess(Constants.Applications.Users) == false)
```

Comment:
This is already handled with the controllers `[Authorize(Policy = AuthorizationPolicies.SectionAccessUsers)]`